cf-custom-resources/lib/trigger-state-machine.js

```diff
@@ -0,0 +1,135 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+"use strict";
+
+const aws = require("aws-sdk");
+
+/**
+ * Main handler, invoked by Lambda
+ */
+exports.handler = async function (event, context) {
+  const responseData = {};
+  const physicalResourceId = event.PhysicalResourceId || event.LogicalResourceId;
+
+  const handler = async function () {
+    switch (event.RequestType) {
+      case "Create":
+      case "Update":
+        const sf = new aws.StepFunctions();
+        const params = {
+          stateMachineArn: event.ResourceProperties.StateMachineARN,
+          input: JSON.stringify({
+            SourceBucket: event.ResourceProperties.SourceBucket,
+            AssetMappingFilePath: event.ResourceProperties.AssetMappingFilePath,
+            DestinationBucket: event.ResourceProperties.DestinationBucket,
+          })
+        };
+
+        const res = await sf.startSyncExecution(params).promise();
+        if (res.status === "FAILED") {
+          throw new Error(`State machine failed: ${res.cause}`);
+        }
+        break;
+      case "Delete":
+        // Do nothing on delete, since this isn't a "real" resource.
+        break;
+      default:
+        throw new Error(`Unsupported request type ${event.RequestType}`);
+    }
+  };
+
+  try {
+    await Promise.race([deadlineExpired(), handler()]);
+    await report(event, context, "SUCCESS", physicalResourceId, responseData);
+  } catch (err) {
+    console.error(`caught error: ${err}`);
+    await report(
+      event,
+      context,
+      "FAILED",
+      physicalResourceId,
+      null,
+      `${err.message} (Log: ${context.logGroupName}/${context.logStreamName})`
+    );
+  }
+};
+
+let deadlineExpired = () => {
+  return new Promise((resolve, reject) => {
+    setTimeout(
+      reject,
+      9 * 60 * 1000 + 30 * 1000 /* 9.5 minutes */,
```

Comment:
interestinggg why 9.5?

Reply:
tbh just copied from another lambda :joy:....i guess i could up it to like 14 minutes

Reply:
ja! The other custom resources time out at like 14.5 minutes I think, leaving half a minute to respond (which is enough)